### PR TITLE
feat(bitmex) - watchTickers

### PR DESCRIPTION
**Pull request by ttodua**
```
const res = await e.watchTickers();

2024-01-31T12:26:43.295Z connecting to wss://ws.bitmex.com/realtime
2024-01-31T12:26:43.743Z onUpgrade
2024-01-31T12:26:43.745Z onOpen
2024-01-31T12:26:43.746Z sending { op: 'subscribe', args: [ 'instrument' ] }
2024-01-31T12:26:43.748Z onMessage {
  info: 'Welcome to the BitMEX Realtime API.',
  version: '2.0.0',
  timestamp: '2024-01-31T12:26:43.840Z',
  docs: 'https://www.bitmex.com/app/wsAPI',
  heartbeatEnabled: false,
  limit: { remaining: 179 }
}
2024-01-31T12:26:43.841Z onMessage {
  success: true,
  subscribe: 'instrument',
  request: { op: 'subscribe', args: [ 'instrument' ] }
}
....
```

res:
```
{
  "SNT/USDT:USDT": {
    symbol: "SNT/USDT:USDT",
    timestamp: 1706704003061,
    datetime: "2024-01-31T12:26:43.061Z",
    high: undefined,
    low: undefined,
    bid: 0.03828,
    bidVolume: undefined,
    ask: 0.03844,
    askVolume: undefined,
    vwap: undefined,
    open: 0.03996,
    close: 0.03996,
    last: 0.03996,
    previousClose: undefined,
    change: 0,
    percentage: 0,
    average: 0.03996,
    baseVolume: 0,
    quoteVolume: 0,
    info: {
      symbol: "SNTUSDT",
      rootSymbol: "SNT",
      state: "Open",
      typ: "FFWCSX",
      listing: "2023-11-08T04:00:00.000Z",
      front: "2023-11-08T04:00:00.000Z",
      positionCurrency: "SNT",
      underlying: "SNT",
      quoteCurrency: "USDT",
      underlyingSymbol: "SNTT=",
      reference: "BMEX",
      referenceSymbol: ".BSNTT",
      maxOrderQty: 1000000000,
      maxPrice: 1000,
      lotSize: 100,
      tickSize: 0.00001,
      multiplier: 1000000,
      settlCurrency: "USDt",
      underlyingToPositionMultiplier: 1,
      quoteToSettleMultiplier: 1000000,
      isQuanto: false,
      isInverse: false,
      initMargin: 0.1,
      maintMargin: 0.05,
      riskLimit: 20000000000,
      riskStep: 20000000000,
      taxed: true,
      deleverage: true,
      makerFee: -0.00015,
      takerFee: 0.00075,
      settlementFee: 0,
      fundingBaseSymbol: ".SNTBON8H",
      fundingQuoteSymbol: ".USDTBON8H",
      fundingPremiumSymbol: ".SNTUSDTPI8H",
      fundingTimestamp: "2024-01-31T20:00:00.000Z",
      fundingInterval: "2000-01-01T08:00:00.000Z",
      fundingRate: 0.0001,
      indicativeFundingRate: 0.0001,
      prevClosePrice: 0.03837,
      prevTotalVolume: 2492200,
      totalVolume: 2492200,
      volume: 0,
      volume24h: 0,
      prevTotalTurnover: 107378725000,
      totalTurnover: 107378725000,
      turnover: 0,
      turnover24h: 0,
      homeNotional24h: 0,
      foreignNotional24h: 0,
      prevPrice24h: 0.03996,
      lastPrice: 0.03996,
      lastPriceProtected: 0.03996,
      lastTickDirection: "PlusTick",
      lastChangePcnt: 0.0278,
      bidPrice: 0.03828,
      midPrice: 0.03836,
      askPrice: 0.03844,
      hasLiquidity: false,
      openInterest: 2600,
      openValue: 99736000,
      fairMethod: "FundingRate",
      fairBasisRate: 0.1095,
      fairBasis: 0,
      fairPrice: 0.03836,
      markMethod: "FairPrice",
      markPrice: 0.03836,
      indicativeSettlePrice: 0.03836,
      instantPnl: false,
      minTick: 0.00001,
      timestamp: "2024-01-31T12:26:43.061Z",
    },
  },

..........

```

### CODE REVIEW

**Comment by ttodua** [comment]
ah, I didn't touch other parts (safeX methods), I will get to them separately in one go in near future.

File: ts/src/pro/bitmex.ts [hunk 1]
```diff
@@ -78,6 +78,41 @@ export default class bitmex extends bitmexRest {
...
+        } else {
+            messageHashes.push (name);
+        }
+        const request = {
+            'op': 'subscribe',
+            'args': messageHashes,
+        };
+        const tickers = await this.watchMultiple (url, messageHashes, this.extend (request, params), messageHashes);
+        if (this.newUpdates) {
+            return tickers;
```
Comment (carlosmiei) [inline ts/src/pro/bitmex.ts:111]: this return type is wrong

Comment (ttodua) [reply ts/src/pro/bitmex.ts]: wait, when subscribing to all instruments, it was correctly returning.

**Comment by carlosmiei** [comment]
DEMO

```
Python v3.11.5
CCXT v4.2.29
bitmex.watchTickers()
{'.100BONKBON': {'ask': None,
                 'askVolume': None,
                 'average': 0.0003,
                 'baseVolume': None,
                 'bid': None,
                 'bidVolume': None,
                 'change': 0.0,
                 'close': 0.0003,
                 'datetime': '2024-01-31T14:38:15.579Z',
                 'high': None,
                 'info': {'deleverage': False,
                          'hasLiquidity': False,
                          'instantPnl': False,
                          'isInverse': False,
                          'isQuanto': False,
```
```
Python v3.11.5
CCXT v4.2.29
bitmex.watchTickers(['BTC/USDT', 'LTC/USDT:USDT'])
{'ask': 289986.0,
 'askVolume': None,
 'average': 48989.5,
 'baseVolume': 0.0271,
 'bid': 48990.0,
 'bidVolume': None,
 'change': 1.0,
 'close': 48990.0,
 'datetime': '2024-01-31T14:38:40.796Z',
 'high': 289987.0,
 'info': {'askPrice': 289986.0,
          'bidPrice': 48990.0,
          'deleverage': False,
          'foreignNotional24h': 6243.96365,
          'hasLiquidity': False,
          'highPrice': 289987.0,
          'homeNotional24h': 0.0271,
          'impactAskPrice': 289986.0,
          'impactBidPrice': 48990.0,
          'impactMidPrice': 169488.0,
          'initMargin': 1.0,
          'instantPnl': False,
          'isInverse': False,
```

File: ts/src/pro/bitmex.ts (hunk 1 above)
Comment (ttodua) [reply ts/src/pro/bitmex.ts]: probably i missed to check with input symbols array. i'll get to that, don't spend time on it, i missed that nuance.

Comment (carlosmiei) [reply ts/src/pro/bitmex.ts]: np I fixed it already

**Pull request merged**